Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -212,17 +254,34 @@ function Enable-StorageEncryptionInTransit
             break
         }
     }
-  
-    #Connect to AzAccount
-    $isContextSet = Get-AzContext
 
-    if ([string]::IsNullOrEmpty($isContextSet))
-    {  
+    if ($UseSystemAssignedManagedIdentity)
+    {
         Write-Host $([Constants]::SingleDashLine)    
         Write-Host "Connecting to AzAccount..."
-        Connect-AzAccount -ErrorAction Stop
+        $currentContext = Connect-AzAccount -Identity -ErrorAction Stop
         Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
     }
+    elseif ($UseUserAssignedManagedIdentity)
+    {
+        Write-Host $([Constants]::SingleDashLine)    
+        Write-Host "Connecting to AzAccount..."
+        $currentContext = Connect-AzAccount -Identity -AccountId $ClientId -ErrorAction Stop
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        #Connect to AzAccount
+        $isContextSet = Get-AzContext
```

Comment:
Should we keep this as first check so that if there is already login context we should continue with that?

Reply:
Yes, updated the condition to check current logged in account.